jetty-nosql/src/main/java/org/eclipse/jetty/nosql/mongodb/MongoUtils.java

```diff
@@ -69,23 +69,23 @@ else if (valueToDecode instanceof DBObject)
             throw new IllegalStateException(valueToDecode.getClass().toString());
         }
     }
-   
-    
-    
+
     public static String decodeName(String name)
     {
-        return name.replace("%2E",".").replace("%25","%");
+        String cleaned = name;
+        cleaned = StringUtil.replace(cleaned, "%2E", ".");
+        cleaned = StringUtil.replace(cleaned, "%25", "%");
```

Comment:
hmmm would it be more efficient to write a method like:
```
   cleaned = StringUtil.replace(cleaned, "%2E", ".", "%25", "%");
```
or if really important, then 
```
    static Trie<String> replacements = StringUtil.createReplacementTrie("%2E", ".", "%25", "%");
    ...
    cleaned = StringUtil.replace(cleaned, replacements);
```
because tries have good getBest semantic

Reply:
hmmm turns out that trie doesn't really have the right API for this, as it returns the value and not a key,value pair, so you don't know what was matched!  Could be worked around with a complex value.... but not sure it is worth it as we have few multi-replacement.

Another idea probably not worth implementing as we have few long search strings would be `StringUtil.replace(String,String,SearchPattern)`

Reply:
I would be easy enough to just have one of the following syntaxes ...

* `StringUtil.replace(String, String[], String[])` <-- my leading preference
* `StringUtil.replace(String, String[][])`  <-- my secondary preference
* `StringUtil.replace(String, Map<String,String>)`  <-- works, but I'm not happy with creating a heavyweight object for this.

Reply:
This specific example, I would probably do as ...

``` java
  cleaned = URIUtil.decodeSpecific(cleaned, ".%");
```

Signature ...

``` java
String URIUtil.decodeSpecific(String input, String charsToDecode);
```

and the balanced method that could be used in the same MongoUtils class for this would be ...

``` java
String URIUtil.encodeSpecific(String input, String charsToEncode);
```

Reply:
that works for me!

Reply:
Done. needs review